apps/web/src/utils/authentication/keys.ts

```diff
@@ -0,0 +1,205 @@
+import { Ed25519Keypair as SuiPair } from '@mysten/sui.js';
+import { Keypair as SolPair } from '@solana/web3.js';
+import { generateSecretKey, InMemorySigner } from '@taquito/signer';
+import { generateID } from '@tkey/common-types';
+import type { UnknownObject } from '@walless/core';
+import { Networks } from '@walless/core';
+import { encryptWithPasscode } from '@walless/crypto';
+import { modules } from '@walless/ioc';
+import type { PrivateKeyDocument, PublicKeyDocument } from '@walless/store';
+import { generateMnemonic, mnemonicToSeedSync } from 'bip39';
+import { decode } from 'bs58';
+import { derivePath } from 'ed25519-hd-key';
+import { SeedPhraseFormatType } from 'utils/w3a';
+import { key } from 'utils/w3a';
+
+const defaultPrefixDerivationPaths: Array<{
+	path: string;
+	network: Networks;
+}> = [
+	{
+		path: "44'/501'",
+		network: Networks.solana,
+	},
+	{
+		path: "44'/784'",
+		network: Networks.sui,
+	},
+	{
+		path: "44'/1729'",
+		network: Networks.tezos,
+	},
+];
+
+export const initBySeedPhraseModule = async (passcode: string) => {
+	let seedPhrases = await key.modules.seedPhraseModule.getSeedPhrases();
+	if (seedPhrases.length === 0) {
+		await key.modules.seedPhraseModule.setSeedPhrase(
+			SeedPhraseFormatType.PRIMARY,
+			generateMnemonic(),
+		);
+		seedPhrases = await key.modules.seedPhraseModule.getSeedPhrases();
+	}
+
+	const writePromises: Promise<UnknownObject>[] = [];
+
+	for (let i = 0; i < seedPhrases.length; i++) {
+		const storedSeed = seedPhrases[i];
+		const rootSeed = mnemonicToSeedSync(storedSeed.seedPhrase);
+
+		defaultPrefixDerivationPaths.forEach(async (prefixObject) => {
```

Comment:
avoid using async await in HOC, could it be replace with `Promise.all`?

Reply:
smell like we should debug async/await life cycle here